src/encoding/text.rs

```diff
@@ -133,8 +169,14 @@ impl DescriptorEncoder<'_> {
             unit,
             const_labels: self.labels,
             family_labels: None,
+            timestamp: self.timestamp,
         })
     }
+
+    fn with_timestamp(mut self) -> Result<Self, SystemTimeError> {
+        self.timestamp = Some(UnixTimestamp::now()?);
```

Comment:
I am confused. For backfilling, you would need to be able to control the timestamp, no? I.e. set it to some time in the past. Here you just set it to the current time.

Reply:
Thanks for having a look. Yes, my original intent for this interface was that you'd always capture the current values, as if producing a frozen exposition. I think it would make sense to accept the timestamp as an argument though.